src/main/java/com/ning/maven/plugins/dependencyversionscheck/AbstractDependencyVersionsMojo.java

```diff
@@ -241,6 +245,18 @@
     /** Keeps track of the longest name for an artifact for printing out nicely. */
     protected int maxLen = -1;
 
+    private final ExecutorService executorService = Executors.newFixedThreadPool(Runtime.getRuntime().availableProcessors() * 5, new ThreadFactory() {
```

Comment:
I think it's fine to have the threads be daemon threads just to be safe, but pool should be shutdown at the end of the `execute()` method, as the JVM running Maven could be around for a long time.

Regarding the object lifecycle of a `Mojo`, I'm having trouble finding explicit documentation on it, but based on how they are used, it seems the only reasonable way is that they are created once per execution (so `execute()` would be called exactly once).